esmvalcore/preprocessor/_derive/toz.py

```diff
@@ -19,16 +22,88 @@
 DOBSON_UNIT = cf_units.Unit('2.69e20 m^-2')
 
 
+def ensure_correct_lon(o3_cube, ps_cube=None):
+    """Ensure that ``o3`` cube contains ``longitude`` and adapt ``ps`` cube."""
+    if o3_cube.coords('longitude'):
+        return (o3_cube, ps_cube)
+
+    # Get zonal mean ps if necessary
+    if ps_cube is not None:
+        ps_cube = ps_cube.collapsed('longitude', iris.analysis.MEAN)
+        ps_cube.remove_coord('longitude')
+
+    # Add longitude dimension to o3 (and ps if necessary) with length 1
+    cubes = (o3_cube, ps_cube)
+    new_cubes = []
+    lon_coord = iris.coords.DimCoord([180.0], bounds=[[0.0, 360.0]],
+                                     var_name='lon',
+                                     standard_name='longitude',
+                                     long_name='longitude',
+                                     units='degrees_east')
+    for cube in cubes:
+        if cube is None:
+            new_cubes.append(None)
+            continue
+        new_dim_coords = [(c, cube.coord_dims(c)) for c in cube.dim_coords]
+        new_dim_coords.append((lon_coord, cube.ndim))
+        new_aux_coords = [(c, cube.coord_dims(c)) for c in cube.aux_coords]
+        new_cube = iris.cube.Cube(cube.core_data()[..., None],
+                                  dim_coords_and_dims=new_dim_coords,
+                                  aux_coords_and_dims=new_aux_coords)
+        new_cube.metadata = cube.metadata
+        new_cubes.append(new_cube)
+
+    return tuple(new_cubes)
+
+
+def interpolate_hybrid_plevs(cube):
+    """Interpolate hybrid pressure levels."""
+    # Use CMIP6's plev19 target levels (in Pa)
+    target_levels = [
+        100000.0,
+        92500.0,
+        85000.0,
+        70000.0,
+        60000.0,
+        50000.0,
+        40000.0,
+        30000.0,
+        25000.0,
+        20000.0,
+        15000.0,
+        10000.0,
+        7000.0,
+        5000.0,
+        3000.0,
+        2000.0,
+        1000.0,
+        500.0,
+        100.0,
+    ]
```

Comment:
these should really be loaded from the CMIP6_coords CMOR table, not hardwired here

Reply:
Done in https://github.com/ESMValGroup/ESMValCore/pull/2509/commits/c344b95f408bae7f6486fd164088319feaf923f8

Reply:
Outstanding, private first class 😁